include/aspect/particle/particle_handler.h

```diff
@@ -0,0 +1,258 @@
+
+        /**
+         * Access to particles in this handler.
+         * TODO: This function eventually needs to go, to not expose internal structure.
+         * This can only be done once World no longer uses this function.
+         */
+        std::multimap<types::LevelInd, Particle<dim,spacedim> > &
+        get_particles();
+
+        /**
+         * Const access to particles in this world.
+         * TODO: This function needs to go to not expose internal structure.
+         * This can only be done once World no longer uses this function.
+         */
+        const std::multimap<types::LevelInd, Particle<dim,spacedim> > &
+        get_particles() const;
+
+        /**
+         * Return the total number of particles in the simulation. Note that
+         * this function does
+         * not compute the number of particles, because that is an expensive
+         * global MPI operation. Instead it returns the number, which is
+         * updated internally every time it might change by a call to
+         * update_n_global_particles().
+         *
+         * @return Total number of particles in simulation.
+         */
+        types::particle_index n_global_particles() const;
+
+        /**
+         * Return the number of particles in the local part of the
+         * triangulation.
+         */
+        types::particle_index n_locally_owned_particles() const;
+
+        /**
+         * Return the number of particles in the given cell.
+         */
+        unsigned int n_particles_in_cell(const typename Triangulation<dim>::active_cell_iterator &cell) const;
+
+        /**
+         * Serialize the contents of this class.
+         */
+        template <class Archive>
+        void serialize (Archive &ar, const unsigned int version);
+
+      private:
+        /**
+         * Address of the triangulation to work on.
+         */
+        SmartPointer<const parallel::distributed::Triangulation<dim,spacedim>,ParticleHandler<dim,spacedim> > triangulation;
+
+        /**
+         * MPI communicator.
+         */
+        MPI_Comm mpi_communicator;
+
+        /**
+         * Set of particles currently in the local domain, organized by
+         * the level/index of the cell they are in.
+         */
+        std::multimap<types::LevelInd, Particle<dim,spacedim> > particles;
+
+        /**
+         * This variable stores how many particles are stored globally. It is
+         * calculated by update_n_global_particles().
+         */
+        types::particle_index global_number_of_particles;
+
+        /**
+         * The maximum number of particles per cell in the global domain. This
+         * variable is important to store and load particle data during
+         * repartition and serialization of the solution. Note that the
+         * variable is only updated when it is needed, e.g. before or after
+         * serialization (before/after mesh refinement, before creating a
+         * checkpoint and after resuming from a checkpoint).
```

Comment:
the opening `(` is placed oddly.

Is the variable not updated upon moving particles around?